pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -195,6 +175,22 @@ func (cl *Client) Run(stop <-chan struct{}) {
 
 	cl.stop = stop
 
+	if !kube.WaitForCacheSync(stop, cl.crdWatcher.HasSynced) {
+		cl.logger.Errorf("Failed to sync Pilot K8S CRD watcher")
+		return
+	}
+	for _, s := range cl.schemas.All() {
+		// From the spec: "Its name MUST be in the format <.spec.name>.<.spec.group>."
+		name := fmt.Sprintf("%s.%s", s.Plural(), s.Group())
+		if s.IsBuiltin() {
+			handleCRDAdd(cl, name)
+		} else {
+			if !cl.crdWatcher.Known(s.GroupVersionResource()) {
+				cl.logger.Warnf("Skipping CRD %v as it is not present", s.GroupVersionKind())
```

Comment:
The log skipping is confusing, maybe CRD xx is not yet present, will be handled latter once it is created